source/common/upstream/upstream_impl.cc

```diff
@@ -272,6 +272,7 @@ HostImpl::createConnection(Event::Dispatcher& dispatcher, const ClusterInfo& clu
       cluster.transportSocketFactory().createTransportSocket(transport_socket_options),
       connection_options);
   connection->setBufferLimits(cluster.perConnectionBufferLimitBytes());
+  cluster.createNetworkFilterChain(*connection);
```

Comment:
Quick clarification from experts: is this connection encrypted or not? For our use case, we want to access the tunneled connection, not the wrapping TLS connection. I think the encryption business is factored into transport socket, so my understanding is that this is the raw connection (h2/http1.1 for HTTP hosts).

Reply:
The underlying transport socket (ssl or raw) is defined by cluster. https://www.envoyproxy.io/docs/envoy/latest/api-v2/api/v2/cds.proto#cluster  search for tls_context and transport_socket

Generally speaking, the downstream decide the cluster name and the cluster config of the selected cluster decide the encryption or not.

Reply:
You will see plain text here, encryption/tap, etc. happens once data is actually written to the transport socket.

Reply:
Thanks, that confirms my expectation.